tlv/record_type_test.go

```diff
@@ -0,0 +1,96 @@
+package tlv
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	fakeCsvDelayType = 1
+
+	fakeIsCoolType = 2
+)
+
+type fakeWireMsg struct {
+	CsvDelay RecordT[TlvType1, uint16]
+
+	IsCool RecordT[TlvType2, bool]
+}
+
+// TestRecordTFromPrimitive tests the RecordT type. We should be able to create
+// types of both record types, and also primitive types, and encode/decode them
+// as normal.
+func TestRecordTFromPrimitive(t *testing.T) {
+	t.Parallel()
+
+	wireMsg := fakeWireMsg{
+		CsvDelay: NewPrimitiveRecord[TlvType1](uint16(5)),
+		IsCool:   NewPrimitiveRecord[TlvType2](true),
```

Comment:
😍

Reply:
Yeah I sorta walked backwards from this type of interface to where things are now. If we don't care about [this](https://github.com/lightningnetwork/lnd/pull/8121#discussion_r1419500523), then we can take the entire `uint16` space.